code/game/gamemodes/objective.dm

```diff
@@ -872,3 +879,1060 @@ var/global/list/possible_items_special = list()
+						return 1
+
+/datum/objective/crew/service/bartender/update_explanation_text()
+	..()
+	explanation_text = "Have [amt] units of [drinkname] on you at shift end."
+
+//------ Have x types of y plants at roundend
+/datum/objective/crew/service/collectplants
+	jobs = "Botanist"
+	var/amt = 0
+	var/list/plants = list()
+
+/datum/objective/crew/service/collectplants/New(text, themind)
+	..()
+	amt = rand(70,100)
+	var/loop = rand(3,5)
+	var/list/blacklist = list(/obj/item/weapon/reagent_containers/food/snacks/grown/mushroom/walkingmushroom)
+	var/list/possibleplants = (typesof(/obj/item/weapon/reagent_containers/food/snacks/grown) - /obj/item/weapon/reagent_containers/food/snacks/grown) + (typesof(/obj/item/weapon/grown) - /obj/item/weapon/grown) - blacklist
+	for(var/i in 1 to loop)
+		var/path = "[pick_n_take(possibleplants)]"
+		plants[path] = round(amt/loop)
+
+/datum/objective/crew/service/collectplants/check_completion()
+	if(owner && owner.current)
+		var/mob/M = owner.current
+		for(var/obj/item/weapon/storage/bag/plants/P in M.contents)
+			for(var/obj/item/I in P)
+				var/path = "[I.type]"
+				if(path in plants)
+					plants[path]--
+		for(var/i in plants)
+			if(plants[i])
+				return 0
+		..(1)
+		return 1
+
+/datum/objective/crew/service/collectplants/update_explanation_text()
+	..()
+	explanation_text = "Collect atleast "
+	for(var/i in 1 to plants.len)
+		var/obj/item/plant = text2path(plants[i])
+		explanation_text += "[plants[plants[i]]] [initial(plant.name)]"
+		if(copytext(explanation_text, -1) != "s")
+			explanation_text += "s"
+		if(i != plants.len)
+			explanation_text += ", "
+	explanation_text += " and keep them in a plant bag at shift end."
+
+//ensure that x areas is clean
+/datum/objective/crew/service/clean
+	jobs = "Janitor"
+	var/list/areas = list()
+
+/datum/objective/crew/service/clean/New(text, themind)
+	..()
+	var/list/possibleareas = list()
+	for(var/A in teleportlocs) // a list with all station areas plus space
+		if(istype(teleportlocs[A], /area/space))
+			continue
+		possibleareas |= teleportlocs[A]
+	for(var/i in 1 to rand(1,2))
+		areas |= pick_n_take(possibleareas)
+
+/datum/objective/crew/service/clean/check_completion()
+	for(var/area/A in areas)
+		for(var/obj/effect/decal/cleanable/C in area_contents(A))
+			return 0
+	..(1)
+	return 1
+
+/datum/objective/crew/service/clean/update_explanation_text()
+	..()
+	explanation_text = "Ensure that "
+	for(var/i in 1 to areas.len)
+		var/area/A = areas[i]
+		explanation_text += "[A.name]"
+		if(i != areas.len)
+			explanation_text += " and " //since it's either 1 or 2.
+	explanation_text += " are completely clean when the shift ends. Please ask for permission before entering in certain places!"
+
+//CIVILIAN! holy shit i didn't believe i'd reach this
+/datum/objective/crew/civilian
+	mytag = "Civilian"
+	jobs = "Captain#Librarian#Lawyer#Chaplain#Clown#Mime#Assistant"
+
+//keep dat fukken disk secure
+/datum/objective/crew/civilian/datfukkendisk
```

Comment:
cause every head was in his own department, but making one department just for captain felt retarded so i moved him to civilian